File: excel2json/chart.py
import openpyxl
import pprint
import json


def toChart(file,sheet,target, legend):
    wb = openpyxl.load_workbook(file)
    sheet = wb[sheet]
    chartData = [['x', 'data', 'legend']]

    for i in range(2, 26):
        item = []
        for j in range(2):
            item.append(sheet.cell(row=i, column=j + 1).value)
        item[0] = item[0].strftime('%H:%M:%S')
        chartData.append(item + [legend])

    pprint.pprint(chartData)

    chart = open(target, 'w')
    chart.write(json.dumps(chartData))


toChart('./chart/chart.xlsx','Sheet1','./chart/chart1.json','l2')

File: excel2json/chart.js
const fs = require('fs');
const Excel = require('exceljs');
const wb = new Excel.Workbook();

function toChart(file, sheet, target, lengend = 'l1') {
    wb.xlsx.readFile(file).then(() => {
        const ws = wb.getWorksheet(sheet);
        const listData = ['x', 'data', 'legend'];
        const max_columns = ws.getColumn(1).values.length;

        for (let i = 2; i < max_columns; i++) {
            let item = [];
            ws.getRow(i).values.forEach((v, i) => {
                if (i === 1) {
                    item.push(new Date(v).toISOString().split('T')[1].slice(0, 5));
                } else if (i > 1) {
                    item.push(v);
                }
            });
            listData.push([...item, lengend]);
        }

        console.log(listData);

        fs.writeFile(target, JSON.stringify(listData), (err) => {
            if (err) throw err;
            // console.log('文件已保存！');
        });
    });
}

toChart('./chart/chart.xlsx', 'Sheet1', './chart/chart1.json');